auslib/db.py

```diff
@@ -720,30 +726,36 @@ def forInsert(self, insertedKeys, columns, changed_by):
             columns[name] = insertedKeys[i]
 
         ts = getMillisecondTimestamp()
-        queries.append(self._insertStatement(changed_by=changed_by, timestamp=ts - 1, **primary_key_data))
-        queries.append(self._insertStatement(changed_by=changed_by, timestamp=ts, **columns))
+        query, _ = self._insertStatement(changed_by=changed_by, timestamp=ts - 1, **primary_key_data)
+        queries.append(query)
+        query, _ = self._insertStatement(changed_by=changed_by, timestamp=ts, **columns)
+        queries.append(query)
         return queries
 
     def forDelete(self, rowData, changed_by):
         """Deletes cause a single row to be created, which only contains the
            primary key data. This represents that the row no longer exists."""
         row = {}
-        for k in rowData:
-            row[str(k)] = rowData[k]
+        table_row_data = {k: rowData[k] for k in rowData.keys() if k in self.table.c}
```

Comment:
Spots like this should probably raise an error eventually too - anytime a caller is trying to do something to a column that doesn't exist, that's a yellow flag.